src/pages/Scholarships.jsx

```diff
@@ -50,6 +50,23 @@ function ScholarshipsPage() {
     history.replace({ search: queryString.stringify(params) });
   };
 
+  const sortBy = params.sortBy ?? DEFAULT_SORT_BY;
+
+  if (
+    sortBy !== 'deadlineAsc' &&
+    sortBy !== 'deadlineDesc' &&
+    sortBy !== 'amountAsc' &&
+    sortBy !== 'amountDesc'
+  ) {
+    pruneQueryParam('sortBy');
+  }
+
+  const key = params.sortBy !== undefined ? sortBy : DEFAULT_SORT_BY;
```

Comment:
This variable isn't necessary, you already made sure `sortBy` is valid so just use `sortBy`. (You could rename it to `sortKey` or something more readable though if it helps)

Reply:
Well when its pruned is when the issue comes because now it no longer is valid meaning it is now `undefined`

Reply:
If this is what you mean then it does not work
```javascript
  if (
    !sortBy.hasOwnProperty('deadlineAsc') &&
    !sortBy.hasOwnProperty('deadlineDesc') &&
    !sortBy.hasOwnProperty('amountAsc') &&
    !sortBy.hasOwnProperty('amountDesc')
  ) {
    pruneQueryParam('sortBy');
  }

  const sortField = sortOptions[sortBy].field;
  const sortDir = sortOptions[sortBy].dir;
```

Reply:
Let me ask you two questions. What is the type of `sortBy`? And then, what kind of object should `hasOwnProperty` be used against? Does it make any sense to check if a string has a property/key of a string?

One of the reasons I strongly dislike JavaScript is because it allows to write things like `"someString".hasOwnProperty("foo")` and it's still valid JavaScript.

Think about it a little more, is `sortBy` the dictionary you're checking in or the key you're checking for?

Reply:
Ok makes sense then I think the code below is what you mean
```javascript
if (!sortOptions.hasOwnProperty('deadline.asc')) {
    pruneQueryParam('sortBy');
}
```

Reply:
Its causing lint issues so should it be something like this?
if (!(sortBy in sortOptions)) {
    pruneQueryParam('sortBy');
}

Reply:
Makes sense, thanks for proactively addressing the lint issue!